nncf/experimental/torch/fx/quantization/quantize_model.py

```diff
@@ -0,0 +1,125 @@
+from nncf.experimental.torch.fx.transformations import revert_quantization_transformations
+from nncf.parameters import ModelType
+from nncf.parameters import QuantizationMode
+from nncf.parameters import TargetDevice
+from nncf.quantization.advanced_parameters import AdvancedQuantizationParameters
+from nncf.quantization.advanced_parameters import QuantizationParameters
+from nncf.quantization.algorithms.post_training.algorithm import PostTrainingQuantization
+from nncf.scopes import IgnoredScope
+
+DEFAULT_RANGE_TYPE = "mean_min_max"
+
+
+def quantize_impl(
+    model: torch.fx.GraphModule,
+    calibration_dataset: Dataset,
+    mode: Optional[QuantizationMode] = None,
+    preset: Optional[QuantizationPreset] = None,
+    target_device: TargetDevice = TargetDevice.ANY,
+    subset_size: int = 300,
+    fast_bias_correction: bool = True,
+    model_type: Optional[ModelType] = None,
+    ignored_scope: Optional[IgnoredScope] = None,
+    advanced_parameters: Optional[AdvancedQuantizationParameters] = None,
+) -> torch.nn.Module:
+    """
+    Implementation of the `quantize()` method for the Torch FX backend.
+    """
+    nncf_logger.warning(
+        "Experimental Torch FX quantization backend is being used for the given torch.fx.GraphModule model."
+        " Torch FX PTQ is an experimental feature, consider using Torch or OpenVino PTQ backends"
+        " in case of errors or a poor model performance."
+    )
+    if fast_bias_correction is False:
+        raise ValueError(f"fast_bias_correction={fast_bias_correction} is not supported")
+    if target_device == TargetDevice.CPU_SPR:
+        raise nncf.InternalError("target_device == CPU_SPR is not supported")
+    if mode is not None:
+        raise ValueError(f"mode={mode} is not supported")
+
+    original_graph_meta = model.meta
+
+    copied_model = deepcopy(model)
+
+    if advanced_parameters is None:
+        advanced_parameters = AdvancedQuantizationParameters()
+    # Default quantization mode is asymmetric
+    activations_quantization_params = advanced_parameters.activations_quantization_params
+    if activations_quantization_params is None:
+        activations_quantization_params = QuantizationParameters()
+        activations_quantization_params.mode = QuantizationScheme.ASYMMETRIC
+        advanced_parameters.activations_quantization_params = activations_quantization_params
+
+    quantization_algorithm = PostTrainingQuantization(
+        preset=preset,
+        target_device=target_device,
+        subset_size=subset_size,
+        fast_bias_correction=fast_bias_correction,
+        model_type=model_type,
+        ignored_scope=ignored_scope,
+        advanced_parameters=advanced_parameters,
+    )
+
+    # BatchNorm operations have 3 output ports,
+    # to make it easier for alorithms to work
+    # with the target graph BatchNorm operations
+    # are being fused
+    _fuse_conv_bn_(copied_model)
```

Comment:
I suggest to move  `_fuse_conv_bn_` transformation into `apply_quantization_transformations` function as part of quantization transformations.

Reply:
Done